chain/src/view.rs

```diff
@@ -150,11 +164,32 @@ pub trait StateWriteExt: StateWrite {
         self.put(state_key::chain_params().into(), params)
     }
 
+    /// Schedules an epoch change for the next block
+    async fn schedule_epoch_change(&mut self) -> Result<()> {
+        let change_height = self.get_block_height().await? + 1;
+
+        let current_epoch = self.epoch().await?;
+        let new_epoch = Epoch {
+            index: current_epoch.index + 1,
+            start_height: change_height,
+        };
+
+        tracing::info!(%change_height, "scheduling epoch change at next block");
+
+        self.put(state_key::epoch_change_at_height(change_height), new_epoch);
+
+        Ok(())
+    }
```

Comment:
Do we want to be scheduling epoch changes and committing the schedule to the chain state, rather than just having the `App` suddenly decide to roll over the epoch? What is going to be consuming the schedule?

Reply:
[1fca310](https://github.com/penumbra-zone/penumbra/pull/2215/commits/1fca310e1abcd78daf123fe599950b9d99654ae6) changed this to using the `App` to roll over the epoch